File: python-systempromptleakage/secure9.py
from flask import Flask, request, jsonify
from openai import OpenAI

app = Flask(__name__)
client = OpenAI()

system_prompt = "You are a code review assistant. Review the provided code and suggest improvements."

@app.route("/review", methods=["POST"])
def review():
    code = request.json.get("code")
    response = client.chat.completions.create(
        model="gpt-4o",
        max_tokens=1024,
        messages=[
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": f"Review this code:\n{code}"}
        ]
    )
    llm_output = response.choices[0].message.content
    return jsonify({"review": llm_output})

File: typescript-excessiveagency/secure9.ts
import OpenAI from "openai";

const openai = new OpenAI();

async function classifyText(userText: string): Promise<string> {
  const response = await openai.chat.completions.create({
    model: "gpt-4o",
    messages: [
      {
        role: "system",
        content: "Classify the sentiment of the provided text as positive, negative, or neutral.",
      },
      {
        role: "user",
        content: userText,
      },
    ],
  });
  return response.choices[0].message.content ?? "";
}
